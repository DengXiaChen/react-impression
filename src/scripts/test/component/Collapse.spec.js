import React from 'react';
import { mount, shallow } from 'enzyme';
import { Collapse } from '../../components/impression';

const setup = () => {
    const props = {
        active: true,
    };

    const wrapper = mount(
        <Collapse {...props}>
            <Collapse.Title>
                Title
            </Collapse.Title>
            <Collapse.Body>
                <div>1</div>
                <div>2</div>
                <div>3</div>
            </Collapse.Body>
        </Collapse>
    );

    const wrapper1 = shallow(
        <Collapse {...props}>
            <Collapse.Title>
                Title
            </Collapse.Title>
            <Collapse.Body>
                <div>1</div>
                <div>2</div>
                <div>3</div>
            </Collapse.Body>
        </Collapse>
    );

    return {
        wrapper,
        props,
        wrapper1,
    };
};

describe('Collapse', () => {
    const { wrapper1, wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.collapse').exists()).toBe(true);
        expect(wrapper.find('.collapse-title').exists()).toBe(true);
        expect(wrapper.find('.collapse-body').exists()).toBe(true);
    });

    it('active', () => {
        expect(wrapper.find('.collapse .active').exists()).toBe(true);
        wrapper.find('.collapse-title').simulate('click');
        expect(wrapper.find('.collapse .active').exists()).toBe(false);
    });

    it('ui', () => {
        expect(wrapper1).toMatchSnapshot();
    });
});
