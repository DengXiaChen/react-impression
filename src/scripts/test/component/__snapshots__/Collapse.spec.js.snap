// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="collapse active"
>
    <CollapseTitle
        onClick={[Function]}
    >
        Title
    </CollapseTitle>
    <CollapseBody
        onClick={[Function]}
    >
        <div>
            1
        </div>
        <div>
            2
        </div>
        <div>
            3
        </div>
    </CollapseBody>
</div>,
  "nodes": Array [
    <div
      className="collapse active"
>
      <CollapseTitle
            onClick={[Function]}
      >
            Title
      </CollapseTitle>
      <CollapseBody
            onClick={[Function]}
      >
            <div>
                  1
            </div>
            <div>
                  2
            </div>
            <div>
                  3
            </div>
      </CollapseBody>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Collapse
        active={true}
>
        <CollapseTitle>
                Title
        </CollapseTitle>
        <CollapseBody>
                <div>
                        1
                </div>
                <div>
                        2
                </div>
                <div>
                        3
                </div>
        </CollapseBody>
</Collapse>,
      "_debugID": 13,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Collapse {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "active": true,
          "children": Array [
            <CollapseTitle>
              Title
</CollapseTitle>,
            <CollapseBody>
              <div>
                            1
              </div>
              <div>
                            2
              </div>
              <div>
                            3
              </div>
</CollapseBody>,
          ],
        },
        "refs": Object {},
        "state": Object {
          "active": true,
        },
        "toggleItemsHandle": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="collapse active"
>
          <CollapseTitle
                    onClick={[Function]}
          >
                    Title
          </CollapseTitle>
          <CollapseBody
                    onClick={[Function]}
          >
                    <div>
                              1
                    </div>
                    <div>
                              2
                    </div>
                    <div>
                              3
                    </div>
          </CollapseBody>
</div>,
        "_debugID": 14,
        "_renderedOutput": <div
          className="collapse active"
>
          <CollapseTitle
                    onClick={[Function]}
          >
                    Title
          </CollapseTitle>
          <CollapseBody
                    onClick={[Function]}
          >
                    <div>
                              1
                    </div>
                    <div>
                              2
                    </div>
                    <div>
                              3
                    </div>
          </CollapseBody>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Collapse
    active={true}
>
    <CollapseTitle>
        Title
    </CollapseTitle>
    <CollapseBody>
        <div>
            1
        </div>
        <div>
            2
        </div>
        <div>
            3
        </div>
    </CollapseBody>
</Collapse>,
}
`;
